Apps/Admin/Client/Components/Details/AccountTab.razor.cs

```diff
@@ -0,0 +1,96 @@
+    using HealthGateway.Admin.Client.Authorization;
+    using HealthGateway.Admin.Client.Store.PatientDetails;
+    using HealthGateway.Admin.Client.Store.PatientSupport;
+    using HealthGateway.Admin.Common.Constants;
+    using HealthGateway.Admin.Common.Models;
+    using HealthGateway.Common.Data.Utils;
+    using HealthGateway.Common.Data.ViewModels;
+    using Microsoft.AspNetCore.Components;
+    using Microsoft.AspNetCore.Components.Authorization;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// Backing logic for the AccountTab component.
+    /// </summary>
+    public partial class AccountTab : FluxorComponent
+    {
+        /// <summary>
+        /// Gets or sets the patient's personal health number.
+        /// </summary>
+        [Parameter]
+        public string Phn { get; set; } = string.Empty;
+
+        [Inject]
+        private IState<PatientSupportState> PatientSupportState { get; set; } = default!;
+
+        [Inject]
+        private IState<PatientDetailsState> PatientDetailsState { get; set; } = default!;
+
+        [Inject]
+        private AuthenticationStateProvider AuthenticationStateProvider { get; set; } = default!;
+
+        [Inject]
+        private IConfiguration Configuration { get; set; } = default!;
+
+        private AuthenticationState? AuthenticationState { get; set; }
+
+        private PatientSupportResult? Patient =>
+            this.PatientSupportState.Value.Result?.SingleOrDefault(x => x.PersonalHealthNumber == this.Phn);
+
+        private DateTime? ProfileCreatedDateTime =>
+            this.Patient?.ProfileCreatedDateTime == null ? null : TimeZoneInfo.ConvertTimeFromUtc(this.Patient.ProfileCreatedDateTime.Value, this.GetTimeZone());
+
+        private DateTime? ProfileLastLoginDateTime =>
+            this.Patient?.ProfileLastLoginDateTime == null ? null : TimeZoneInfo.ConvertTimeFromUtc(this.Patient.ProfileLastLoginDateTime.Value, this.GetTimeZone());
+
+        private IEnumerable<MessagingVerificationModel> MessagingVerifications => this.PatientDetailsState.Value.MessagingVerifications ?? Enumerable.Empty<MessagingVerificationModel>();
+
+        private bool PatientSupportDetailsLoading => this.PatientDetailsState.Value.IsLoading;
+
+        private bool CanViewHdid => this.UserHasRole(Roles.Admin) || this.UserHasRole(Roles.Reviewer);
+
+        private bool CanViewMessagingVerifications => this.UserHasRole(Roles.Admin) || this.UserHasRole(Roles.Reviewer);
+
+        private bool IsGatewayUser => this.Patient?.Status == PatientStatus.Default;
+
+        /// <inheritdoc/>
+        protected override async Task OnInitializedAsync()
+        {
+            await base.OnInitializedAsync();
+            this.AuthenticationState = await this.AuthenticationStateProvider.GetAuthenticationStateAsync();
+        }
+
+        private bool UserHasRole(string role)
+        {
+            return this.AuthenticationState?.User.IsInRole(role) == true;
+        }
```

Comment:
We should consider moving this into a service in the future and introducing a method that checks if the user matches one of several valid roles, since this is being used in many places.

Reply:
I started looking at this, and see it might be a little more complicated with some of the asynchronous calls, I'm going to add it to the team prep, to introduce services to reduce complexity in Blazor.